lib/Provider/MailProvider.php

```diff
@@ -0,0 +1,238 @@
+		}
+		// return list of services for user
+		return $services;
+
+	}
+
+	/**
+	 * retrieve a service with a specific id
+	 *
+	 * @since 2024.05.25
+	 *
+	 * @param string $userId			user id
+	 * @param string $serviceId			service id
+	 *
+	 * @return IService|null			returns service object or null if none found
+	 */
+	public function findServiceById(string $userId, string $serviceId): IService | null {
+
+		// evaluate if id is a number
+		if (is_numeric($serviceId)) {
+			try {
+				// retrieve service details from data store
+				$account = $this->AccountService->find($userId, (int) $serviceId);
+			} catch(\Throwable $th) {
+				return null;
+			}
+		}
+		// evaliate if service details where found
+		if ($account instanceof Account) {
+			// extract values
+			$serviceId = (string) $account->getId();
+			$label = $account->getName();
+			$address = new MailAddress($account->getEmail(), $account->getName());
+			$identity = new MailServiceIdentity();
+			$location = new MailServiceLocation();
+			// return mail service object
+			return (new MailService($this->container, $userId, $serviceId, $label, $address, $identity, $location));
+		}
+
+		return null;
+		
+	}
+
+	/**
+	 * retrieve a service for a specific mail address
+	 *
+	 * @since 2024.05.25
+	 *
+	 * @param string $userId			user id
+	 * @param string $address			mail address (e.g. test@example.com)
+	 *
+	 * @return IService					returns service object or null if none found
+	 */
+	public function findServiceByAddress(string $userId, string $address): IService | null {
+
+		try {
+			// retrieve service details from data store
+			$accounts = $this->AccountService->findByUserIdAndAddress($userId, $address);
+		} catch(\Throwable $th) {
+			return null;
+		}
+		// evaliate if service details where found
+		if (is_array($accounts) && count($accounts) > 0 && $accounts[0] instanceof Account) {
+			// extract values
+			$serviceId = (string) $accounts[0]->getId();
+			$label = $accounts[0]->getName();
+			$address = new MailAddress($accounts[0]->getEmail(), $accounts[0]->getName());
+			$identity = new MailServiceIdentity();
+			$location = new MailServiceLocation();
+			// return mail service object
+			return (new MailService($this->container, $userId, $serviceId, $label, $address, $identity, $location));
+		}
+
+		return null;
+
+	}
+
+	/**
+	 * construct a new empty service object
+	 *
+	 * @since 30.0.0
+	 *
+	 * @return IService					blank service object
+	 */
+	public function initiateService(): IService {
+
+		return (new MailService($this->container));
+
+	}
+
+	/**
+	 * create a service configuration for a specific user
+	 *
+	 * @since 2024.05.25
+	 *
+	 * @param string $userId			user id
+	 * @param IService $service 		service object
+	 *
+	 * @return string					id of created service
+	 */
+	public function createService(string $userId, IService $service): string {
+
+		return '';
+
+	}
```

Comment:
Not implemented? Should this throw an exception instead or be implemented?

Same for the other stubs.